Sources/Instrumentation/Instrument.swift

```diff
@@ -50,7 +50,11 @@ public protocol Instrument {
     ///   - carrier: The `Carrier` that was used to propagate values across boundaries.
     ///   - context: The `BaggageContext` into which these values should be injected.
     ///   - extractor: The `Extractor` that extracts values from the given `Carrier`.
-    func extract<Carrier, Extractor>(_ carrier: Carrier, into context: inout BaggageContext, using extractor: Extractor)
+    func extract<Carrier, Extractor>(
+        _ carrier: Carrier,
+        into context: inout BaggageContextProtocol,
```

Comment:
Mixed feelings on the protocol here tbh.

This is a _low level_ API, no need for convenience here.

Reply:
Yep, correct. I'll change it back to `BaggageContext` and test how it affects interacting with the API.